### PR TITLE
Add WASM CI checks and make availability-store compile for WASM

### PR DESCRIPTION
**Pull request by tomaka**
cc #625 

The change to `availability-store` is kind of the Polkadot equivalent to https://github.com/paritytech/substrate/pull/3957, except that we enforce this at compile time.

In order to prevent this from breaking accidentally, also adds CI checks.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -131,6 +131,23 @@ test-linux-stable:                 &test
     - sccache -s
 
 
+check-web-wasm:                    &test
+  stage:                           test
+  <<:                              *test-refs
+  <<:                              *docker-env
+  <<:                              *compiler_info
```
Comment (tomaka) [inline .gitlab-ci.yml]: I copy-pasted these lines but I have no idea what they are.

**Comment by tomaka** [comment]
Oh. Interestingly, running `cargo build --package foo` fails whereas doing `cd foo && cargo build` works. Will investigate more.

**Comment by rphmeier** [comment]
Why does the availability store need to compile to WASM?

**Comment by tomaka** [comment]
> Why does the availability store need to compile to WASM?

We need some conditional compilation *somewhere*, and where is a question of surprise factor/usability/cleanness.

Either we make `availability-store` compile for WASM, or we have to tweak the Cargo.toml of the node in order to not compile `availability-store`.
Since it's going to be the same for other crates, the code of the node could become quite messy and difficult to read if all the conditional compilation happens there.

**Comment by rphmeier** [comment]
I don't follow, do you mean that this should be used in the runtime? Because especially after #345 that won't be possible

**Comment by tomaka** [comment]
Ah no, this PR is about wasm-in-the-browser! Sorry for the confusion.

**Pull request merged**